File: launchdarkly/node/sample.js
/*
 * Copyright (c) 2024 Gitar, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Example of using the LaunchDarkly JavaScript SDK to evaluate feature flags.
 *
 * https://docs.launchdarkly.com/sdk/client-side/javascript
 */

import * as LaunchDarkly from "@launchdarkly/node-server-sdk";

const SDK_KEY = "<SDK_KEY>";
const context = {
	kind: "user",
	name: "Sandy",
	key: "example-context-key",
};
const client = LaunchDarkly.init(SDK_KEY);

client.once("ready", () => {
	client.variation(
		"is-dark-mode-enabled",
		context,
		false,
		(err, useDarkMode) => {
			if (useDarkMode) {
				console.log("Dark mode is enabled");
			} else {
				console.log("Dark mode is disabled");
			}
		},
	);
});

File: launchdarkly/dart/sample.dart
import 'package:launchdarkly_flutter_client_sdk/launchdarkly_flutter_client_sdk.dart';

class SampleDart {
  final LDClient ldClient;

  SampleDart(String sdkKey) : ldClient = LDClient(sdkKey);

  void sampleMethod1() async {
    LDUser user = LDUser(
      key: "context-key-123abc",
      name: "Sandy",
    );

    await ldClient.start(user);

    bool flagValue = await ldClient.boolVariation("is-dark-mode-enabled", false);

    if (flagValue) {
      // The code to run if the feature is on
      System.out.println("Dark mode!");
    } else {
      // The code to run if the feature is off
      bool otherFlag = await ldClient.boolVariation("is-ameyas-weird-blue-theme", false);
      if (otherFlag) {
        System.out.println("Why? Just... why?");
      } else {
        System.out.println("Light mode it is, then!");
      }
    }

    await ldClient.close();
  }
}

void main() {
  SampleDart sample = SampleDart("sdk-key-123abc");
  sample.sampleMethod1();
}
